### PR TITLE
Port key value parser from Stanza

### PR DESCRIPTION
**Pull request by jsirianni**
Resolves https://github.com/open-telemetry/opentelemetry-log-collection/issues/455

### CODE REVIEW

File: operator/parser/keyvalue/keyvalue.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+			quoted = !quoted
+		}
+		return !quoted && r == ' '
+	})
+	return raw
+}
+
+// remove surrounding quotes and trim leading and trailing space
+func cleanString(input string) string {
+	if len(input) < 1 {
```
Comment (djaglowski) [inline operator/parser/keyvalue/keyvalue.go:143]:
```suggestion
	if input = "" {
```

File: operator/parser/keyvalue/keyvalue.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	parserOperator, err := c.ParserConfig.Build(logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Delimiter == c.PairDelimiter {
+		return nil, errors.New("delimiter and pair_delimiter cannot be the same value")
+	}
+
+	if len(c.Delimiter) == 0 {
```
Comment (djaglowski) [inline operator/parser/keyvalue/keyvalue.go:62]: nit:
```suggestion
	if c.Delimiter == "" {
```

**Review by djaglowski — COMMENTED** [review]
Looks good. Just a couple nits.

**Comment by jsirianni** [comment]
@djaglowski thanks for the review, everything should be sorted out.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**